ecl/hql/hqlexpr.cpp

```diff
@@ -4234,7 +4234,19 @@ void CHqlRealExpression::updateFlagsAfterOperands()
     switch (op)
     {
     case no_pure:
-        infoFlags &= ~(HEFnoduplicate|HEFaction|HEFthrowds|HEFthrowscalar|HEFcontainsSkip);
+        //If pure() has extra parameters they allow flags to be selectively removed
+        if (numChildren() > 1)
+        {
+            ForEachChildFrom(i, this, 1)
+            {
+                IHqlExpression * cur = queryChild(i);
+                IAtom * name = cur->queryName();
+                if (name == throwAtom)
+                    infoFlags &= ~(HEFthrowscalar);
```

Comment:
Should there be an else here (perhaps an error, since the name is not recognized?)

Reply:
yes, I could throw an exception if unrecognised,